src/transform/class/inheritance/Prototypal.js

```diff
@@ -0,0 +1,169 @@
+import {matchesAst, extract} from '../../../utils/matchesAst';
+
+/**
+ * Processes nodes to detect super classes and return information for later
+ * transformation.
+ *
+ * Detects:
+ *   function Class1() {}
+ *   function Class2() {}
+ *   Class1.prototype = Object.create(Class2.prototype);
+ *   Class1.prototype.constructor = Class1;
+ * and:
+ *   function Class1() {}
+ *   function Class2() {}
+ *   Class1.prototype = new Class2();
+ *   Class1.prototype.constructor = Class1;
+ */
+export default class Prototypal {
+  /**
+   * @param {Object} cfg
+   *   @param {PotentialClass[]} cfg.potentialClasses
+   */
+  constructor({potentialClasses}) {
+    this.potentialClasses = potentialClasses;
+    this.prototypeAssignments = [];
+  }
+
+  /**
+   * Process a node and return inheritance details if found.
+   * @param {Object} node
+   * @param {Object} parent
+   * @returns {Object}
+   *            {String}   className
+   *            {Node}     superClass
+   *            {Object[]} replacements
+   */
+  process(node, parent) {
+    var m;
+    if ((m = this.matchPrototypeAssignment(node))) {
+      if (this.potentialClasses[m.className]) {
+        this.prototypeAssignments[m.className] = {
+          node,
+          parent,
+          superClass: m.superClass
+        };
+      }
+    }
+    else if ((m = this.matchConstructorAssignment(node))) {
+      var prototypeAssignment = this.prototypeAssignments[m.className];
+      if (this.potentialClasses[m.className] && prototypeAssignment) {
+        return {
+          className: m.className,
+          superClass: prototypeAssignment.superClass,
+          replacements: [
+            {node, parent, replacements: []},
+            {
+              node: prototypeAssignment.node,
+              parent: prototypeAssignment.parent,
+              replacements: []
+            }
+          ]
+        };
+      }
+    }
+    return null;
+  }
+
+  /**
+   * @param {Object} node
+   * @return {Boolean}
+   */
```

Comment:
This comments doesn't really add much value - it's also incorrect as the return type is `Object/Boolean`.

Would be more helpful to describe what this function matches.